File: behavioral/python/state/Account.py
from AccountA   import AccountA
from TRYAccount import TRYAccount

class Account:
    __account: AccountA = None

    def __init__(self):
        self.__account = TRYAccount()

    def withdrawMoney(self, moneyAmount: int)->None:
        self.__account.withdrawMoney(moneyAmount)
    def depositMoney(self, moneyAmount: int)->None:
        self.__account.depositMoney(moneyAmount)
    def changeAccount(self, account:AccountA)->None:
        if(self.__account != None):
            print("Account change " + self.__account.getAccountName() + " to " + account.getAccountName())
        self.__account = account

File: behavioral/jv/state/src/Account.java
public class Account {
    private AccountA account;
    public Account(){
        this.account = null;
    }
    public void setAccount(AccountA account){
        if(this.account != null)
            System.out.println("Account change " + this.account.getAccountName() + " to " + account.getAccountName());
        else
            System.out.println("Select " + account.getAccountName() + " account.");
        this.account = account;
    }
    public void withdrawMoney(int moneyAmount) {
        if(this.account != null)
            this.account.withdrawMoney(moneyAmount);
    }

    public void depositMoney(int moneyAmount) {
        if(this.account != null)
            this.account.depositMoney(moneyAmount);
    }
}
